prow/cmd/deck/static/prow/prow.ts

```diff
@@ -648,20 +667,34 @@ function redraw(fz: FuzzySearch): void {
         max = 2 * 3600;
     }
     drawJobHistogram(totalJob, jobHistogram, now - (12 * 3600), now, max);
+    if (rerunStatus != null) {
```

Comment:
This is an interesting paradigm change - elsewhere, management of `modal` is _not_ handled by the redraw function, but instead updated directly by callback functions.

While this is a reasonable approach in itself, I'm not sure that it is entirely consistent with how the rest of the page works (or, perhaps, how the rest of the page handles the modal is wrong and it should all be done here - we're basically doing manual react anyway...).

Not sure I'm saying you must change this, but maybe something to think about.

Reply:
On further thought, disregard the above - this is the right way to go.